modules/model/shared/src/main/scala/observe/model/InstrumentDynamicConfig.scala

```diff
@@ -5,59 +5,95 @@ package observe.model
 
 import cats.Eq
 import cats.syntax.either.*
+import cats.syntax.eq.*
 import io.circe.Decoder
 import io.circe.DecodingFailure
 import io.circe.Encoder
 import io.circe.JsonObject
 import io.circe.syntax.*
 import lucuma.core.enums.Instrument
-import lucuma.core.model.sequence.gmos.DynamicConfig
+import lucuma.core.model.sequence.flamingos2.Flamingos2DynamicConfig
+import lucuma.core.model.sequence.gmos
+import lucuma.odb.json.flamingos2.given
 import lucuma.odb.json.gmos.given
 import lucuma.odb.json.time.transport.given
 import lucuma.odb.json.wavelength.transport.given
 
 // The whole purpose of this class is to bundle the instrument with its dynamic configuration,
 // so that the dynamic configuration can be decoded correctly.
-enum InstrumentDynamicConfig(val instrument: Instrument):
-  def config: DynamicConfig
+sealed trait InstrumentDynamicConfig(val instrument: Instrument):
+  type D
 
-  case GmosNorth(config: DynamicConfig.GmosNorth)
-      extends InstrumentDynamicConfig(Instrument.GmosNorth)
-
-  case GmosSouth(config: DynamicConfig.GmosSouth)
-      extends InstrumentDynamicConfig(Instrument.GmosSouth)
+  def config: D
 
 object InstrumentDynamicConfig:
-  given Eq[InstrumentDynamicConfig] = Eq.by(x => (x.instrument, x.config))
+  case class GmosNorth(config: gmos.DynamicConfig.GmosNorth)
+      extends InstrumentDynamicConfig(Instrument.GmosNorth) {
+    type D = gmos.DynamicConfig.GmosNorth
+  }
+
+  case class GmosSouth(config: gmos.DynamicConfig.GmosSouth)
+      extends InstrumentDynamicConfig(Instrument.GmosSouth) {
+    type D = gmos.DynamicConfig.GmosSouth
+  }
+
+  case class Flamingos2(config: Flamingos2DynamicConfig)
+      extends InstrumentDynamicConfig(Instrument.Flamingos2) {
+    type D = Flamingos2DynamicConfig
+  }
+
+  object GmosNorth:
+    given Eq[GmosNorth] = Eq.by(x => (x.instrument, x.config))
```

Comment:
do we need to compare instrument as it is essentially fixed ?

Reply:
Maybe it's not necessary, but I'll leave it for extra caution.